src/main.rs

```diff
@@ -66,7 +66,9 @@ fn start_application() {
     let mut siv = Cursive::new();
     siv.add_global_callback('q', Cursive::quit);
     siv.add_global_callback(Key::Esc, |s| {
-        s.pop_layer();
+        if s.pop_layer().is_none() {
```

Comment:
This won't quit the application after the last layer was removed. If we check if the amount of layers is 0 and quit then, it'll quit correctly
```rust
if s.pop_layer().is_none() || s.screen().len() == 0 {
...
}
```